File: Ad-Hoc/pagina11/2144/2144.cpp
#include <iostream>
#include <vector>

using namespace std;

float repeticoes_maxima(int peso, int reps) {
    return peso * (1 + (float(reps) / 30));  // Garantir que a divisão seja feita com float
}

float media_geral(vector<float>& medias) {
    float soma = 0;  // Inicializar a soma com 0
    for (const float& media : medias)
        soma += media;

    return soma / medias.size();
}

int main() {
    vector<float> medias;
    int w1, w2, reps;

    while (cin >> w1 >> w2 >> reps) {
        if (w1 == 0 && w2 == 0 && reps == 0)
            break;

        float w1_max = repeticoes_maxima(w1, reps);  // Garantir que o cálculo de repetição máxima seja float
        float w2_max = repeticoes_maxima(w2, reps);
        float media_rm = (w1_max + w2_max) / 2.0;

        medias.push_back(media_rm);

        if (1 <= media_rm && media_rm < 13) {
            cout << "Nao vai da nao" << endl;
        } else if (13 <= media_rm && media_rm < 14) {
            cout << "E 13" << endl;
        } else if (14 <= media_rm && media_rm < 40) {
            cout << "Bora, hora do show! BIIR!" << endl;
        } else if (40 <= media_rm && media_rm <= 60) {
            cout << "Ta saindo da jaula o monstro!" << endl;
        } else {
            cout << "AQUI E BODYBUILDER!!" << endl;
        }
    }

    float media = media_geral(medias);

    if (media > 40.0) {
        cout << "" << endl;
        cout << "Aqui nois constroi fibra rapaz! Nao e agua com musculo!" << endl;
    }

    return 0;
}

File: Ad-Hoc/pagina11/2144/2144.java
import java.util.Scanner;
import java.util.ArrayList;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        ArrayList<Float> medias = new ArrayList<>();
        int i = 0;

        while(in.hasNextLine()){

            String data[] = in.nextLine().split(" ");
            int w1 = Integer.parseInt(data[0]);
            int w2 = Integer.parseInt(data[1]);
            int reps = Integer.parseInt(data[2]);

            if(w1 == 0 && w2 == 0 && reps == 0) break;

            float w1_max = rep_maxima(w1, reps);
            float w2_max = rep_maxima(w2, reps);
            float media_rm = (w1_max + w2_max) / 2.0f;

            medias.add(media_rm);

            if (1 <= media_rm && media_rm < 13) {
                System.out.println("Nao vai da nao");
            } else if (13 <= media_rm && media_rm < 14) {
                System.out.println("E 13");
            } else if (14 <= media_rm && media_rm < 40) {
                System.out.println("Bora, hora do show! BIIR!");
            } else if (40 <= media_rm && media_rm <= 60) {
                System.out.println("Ta saindo da jaula o monstro!");
            } else {
                System.out.println("AQUI E BODYBUILDER!!");
            }

            i++;
        }

        float media_geral = calcular_media_geral(medias);

        if(media_geral > 40.0){
            System.out.println();
            System.out.println("Aqui nois constroi fibra rapaz! Nao e agua com musculo!");
        }
        in.close();
    }

    public static float rep_maxima(int peso, int reps){
        return peso * (1 + (reps / 30.0f));
    }

    public static float calcular_media_geral(ArrayList<Float> medias){
        float soma = 0.0f;
        for (float media : medias) {
            soma += media;
        }

        return soma / medias.size();
    }
}
